astropy/io/ascii/ui.py

```diff
@@ -1005,8 +1004,8 @@ def write(
         "write", format=format, fast_writer=fast_writer, overwrite=overwrite, **kwargs
     )
 
-    if isinstance(output, (str, bytes, os.PathLike)):
-        output = os.path.expanduser(output)
+    if isinstance(output, (str, bytes, os.PathLike)) and output:
+        output = Path(output).expanduser()
```

Comment:
I would prefer to revert the changes In this function and stick with the original `os.path.expanduser()`. In this case using `Path` is introducing new complexity to the logic because of the `Path("") => cwd()` behavior. So maybe add a `#noqa ...`.